src/test/java/unstable/LabelPickerTests.java

```diff
@@ -4,207 +4,38 @@
 import javafx.application.Platform;
 import javafx.scene.control.TextField;
 import javafx.scene.input.KeyCode;
-import org.controlsfx.control.NotificationPane;
+
 import org.junit.Test;
+
+import backend.resource.TurboIssue;
 import ui.UI;
-import ui.components.FilterTextField;
 import ui.listpanel.ListPanelCell;
-import util.PlatformEx;
 import util.events.ShowLabelPickerEvent;
-
 import static org.junit.Assert.assertEquals;
-import static ui.components.KeyboardShortcuts.UNDO_LABEL_CHANGES;
 
 public class LabelPickerTests extends UITest {
 
-    @Test
-    public void showLabelPickerTest() {
-        FilterTextField filterTextField = find("#dummy/dummy_col0_filterTextField");
-        click(filterTextField);
-        type("hello");
-        assertEquals("hello", filterTextField.getText());
-        filterTextField.setText("");
-
-        ListPanelCell listPanelCell = find("#dummy/dummy_col0_9");
-        click(listPanelCell);
-
-        Platform.runLater(stage::hide);
-        UI.events.triggerEvent(new ShowLabelPickerEvent(listPanelCell.getIssue()));
-        waitUntilNodeAppears("#labelPickerTextField");
+    private static final String QUERY_FIELD_ID = "#queryField";
+    private static final String DEFAULT_ISSUECARD_ID = "#dummy/dummy_col0_9";
 
-        TextField labelPickerTextField = find("#labelPickerTextField");
+    @Test
+    public void showLabelPicker_typeQuery_displaysCorrectly() {
+        triggerLabelPicker(getIssueCard(DEFAULT_ISSUECARD_ID).getIssue());
+        TextField labelPickerTextField = find(QUERY_FIELD_ID);
         click(labelPickerTextField);
         type("world");
         assertEquals("world", labelPickerTextField.getText());
         push(KeyCode.ESCAPE);
-        waitUntilNodeDisappears("#labelPickerTextField");
+        waitUntilNodeDisappears(QUERY_FIELD_ID);
     }
```

Comment:
Currently it is tested by triggering the LabelPicker from the main stage and uses TestFX for UI related operation. Should we create another issue on writing UI test ? I think there is a need to explore better way to test picker-like UI